tsbridge/config.go

```diff
@@ -1,168 +1,29 @@
-// Copyright 2018 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//      http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-// Package tsbridge deals with Time Series Bridge configuration files and metric representations.
-// This file has code related to configuration files.
 package tsbridge
 
 import (
-	"context"
-	"fmt"
-	"io/ioutil"
-	"os"
 	"time"
-
-	"github.com/google/ts-bridge/datadog"
-	"github.com/google/ts-bridge/influxdb"
-	"github.com/google/ts-bridge/storage"
-
-	log "github.com/sirupsen/logrus"
-	validator "gopkg.in/validator.v2"
-	yaml "gopkg.in/yaml.v2"
 )
 
-// Config is what the YAML configuration file gets deserialized to.
 type Config struct {
-	DatadogMetrics  []*DatadogMetricConfig  `yaml:"datadog_metrics"`
-	InfluxDBMetrics []*InfluxDBMetricConfig `yaml:"influxdb_metrics"`
-
-	StackdriverDestinations []*DestinationConfig `yaml:"stackdriver_destinations"`
-
-	// internal list of metrics that gets populated when configuration file is read.
-	metrics []*Metric
-}
-
-// DestinationConfig defines configuration for a Stackdriver project metrics are written to.
-// Name is only used internally to set destination for a specific imported metric.
-type DestinationConfig struct {
-	Name      string `validate:"nonzero"`
-	ProjectID string `yaml:"project_id" validate:"regexp=^[A-Za-z0-9:.-]*$"`
-}
-
-// SourceMetricConfig defines some common parameters that any imported metric must have, irrespective of the
-// monitoring system data is coming from.
-type SourceMetricConfig struct {
-	Name        string `validate:"regexp=^[A-Za-z0-9]\\w*$"`
-	Destination string `validate:"nonzero"`
-}
-
-// DatadogMetricConfig combines common metric configuration parameters with Datadog-specific ones.
-type DatadogMetricConfig struct {
-	SourceMetricConfig   `yaml:"_,inline"`
-	datadog.MetricConfig `yaml:"_,inline"`
-}
-
-// InfluxDBMetricConfig combines common metric configuration parameters with InfluxDB-specific ones.
-type InfluxDBMetricConfig struct {
-	SourceMetricConfig    `yaml:"_,inline"`
-	influxdb.MetricConfig `yaml:"_,inline"`
-}
-
-// Metrics returns a list of metrics defined in the configuration file.
-func (c *Config) Metrics() []*Metric {
-	return c.metrics
+	Options ConfigOptions
```

Comment:
Does Config need to be a separate struct from ConfigOptions? It seems unnecessary given that one only contains the other.

Reply:
This is intentional as I later want to refactor how storage configuration works, so I'm keeping all flags in the same place.